### PR TITLE
test: refactor web and native tests to pass linting

### PR DESCRIPTION
**Pull request by craigbilner**
* Split out and focus tests
* Add styles to web snapshots
* Fix typo
* Refactor fixtures to be dynamic
* Export minimalise from jest-serializer

### CODE REVIEW

File: packages/article-image/__tests__/shared.native.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    compose(
+      print,
+      flattenStyleTransform,
+      minimalNativeTransform,
+      minimaliseTransform((value, key) => key === "style")
+    )
+  );
+
+  shared(component => {
+    const testInstance = TestRenderer.create(component);
```
Comment (colincclark) [inline packages/article-image/__tests__/shared.native.js]: this is `const tree = renderer.create....` everywhere else

File: packages/article-image/__tests__/shared-with-style.base.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+export default makeTest => {
+  it("renders primary image with caption and credits", () =>
+    expect(
+      makeTest(
+        <ArticleImage
+          imageOptions={primaryImage.imageOptions}
+          captionOptions={primaryImage.captionOptions}
+        />
+      )
+    ).toMatchSnapshot());
```
Comment (colincclark) [inline packages/article-image/__tests__/shared-with-style.base.js]: missing snapshot names

File: packages/article-image/__tests__/shared-with-style.base.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
+import "react-native";
```
Comment (colincclark) [inline packages/article-image/__tests__/shared-with-style.base.js]: don't think this is required

File: packages/article-image/__tests__/shared.native.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  compose,
+  flattenStyleTransform,
+  minimaliseTransform,
+  minimalNativeTransform,
+  print
+} from "@times-components/jest-serializer";
+import TestRenderer from "react-test-renderer";
+import shared from "./shared.base";
+
+jest.mock("@times-components/image", () => ({
```
Comment (colincclark) [inline packages/article-image/__tests__/shared.native.js]: :(

File: packages/article-image/__tests__/android/article-image-with-style.android.test.js [hunk 5]
```diff
@@ -0,0 +1,5 @@
+import shared from "../shared-with-style.native";
+
+describe("Android", () => {
```
Comment (colincclark) [inline packages/article-image/__tests__/android/article-image-with-style.android.test.js]: can we stick with lowercase convention, here and in the cli?

`describe("ios/android/web", () => {`

File: packages/article-image/__tests__/shared.native.js (hunk 1 above)
Comment (craigbilner) [reply packages/article-image/__tests__/shared.native.js]: Will be fixing this across the code base and have added #1058

File: packages/article-image/__tests__/shared-with-style.base.js (hunk 2 above)
Comment (craigbilner) [reply packages/article-image/__tests__/shared-with-style.base.js]: :+1: have added #1059

**Pull request merged**